[11.0][MIG] l10n_es_aeat_mod130

victor-ditay opened the pull request.

Commit:
[ADD] l10n_es_aeat_mod130: Módulo para la declaración del modelo 130 - IRPF pago fraccionado estimación directa

Commit:
Launchpad automatic translations update.

Commit:
[FIX] l10n_es_aeat_mod130: Corrección del archivo exportado para evitar código 0339 en la presentación.

acysos committed:
[FIX] l10n_es_aeat: compatibilidad de la función export_to_boe para el modelo 340

Commit:
[FIX] l10n_es_aeat_mod130: Tilde en archivo que hace fallar runbot.

Commit:
[FIX] l10n_es_aeat_mod130/mod303: import corregido para que valga también para la v8.

rlizana committed:
[MIG] l10n_es_aeat_mo130

* Clases a CapWords
* PEP8
* Icono del modulo

pedrobaeza committed:
[FIX] l10n_es_aeat_mod130: Problema en la búsqueda cuando los periodos no son trimestrales + PEP8 79 cols

Commit:
[IMP] unificar campos en el módulo base + migrar a la nueva api.

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat_*: PEP8 + Refinamientos varios

oca-transbot committed:
OCA Transbot updated translations from Transifex

hugosantosred committed:
[IMP] Añadir reglas de registro multicompañía a informes financieros e informes AEAT

Commit:
[MIG] l10n_es_aeat_mod130: Migration to 11.0

victor-ditay commented:
cc @Tecnativa

pedrobaeza commented:
Está ya listo para revisar?

victor-ditay commented:
Creo que ya esta bastante avanzado y se puede revisar, pero si veo que aun no pasa Travis ni Runbot por lo que ire machacando los errores

pedrobaeza commented:
OK, me espero a que avises que está ya en verde.

victor-ditay commented:
Travis is green but not Runbot.

Help wanted with Runbot, no Warning or Errors inside Full logs

victor-ditay commented:
Fails every time try force rebuild runbot

pedrobaeza commented:
@victormmtorres podrías añadir tests al módulo? Echa un vistazo al del 111 para inspirarte (https://github.com/OCA/l10n-spain/blob/11.0/l10n_es_aeat_mod111/tests/test_l10n_es_aeat_mod111.py):

- Heredar del TestL10nEsAeatModBase que ya crea el plan contable y trae herramientas para rápida creación de facturas.
- Crear alguna factura de compra y de venta declarando el diccionario de impuestos `taxes_purchase` y el de `taxes_sale` que no ves en el 111. Pon los impuestos normales `SIVA21B` y `PIVA21B` como ejemplo (aquí no es lo más importante, ya que se miden ingresos y gastos).
- Crea el modelo.
- Ejecuta el cálculo.
- Comprueba los valores de lo que se debería haber calculado..
- Puedes hacer variantes de marcar la casilla de deducción para ver cómo varía el cálculo.
- Copia de aquí para "testear" la exportación a BOE: https://github.com/OCA/l10n-spain/blob/f5b2f149e357dd2e33e58e30d39dd18a17f357de/l10n_es_aeat_mod303/tests/test_l10n_es_aeat_mod303.py#L348-L361 (lo pongo entre comillas porque lo que estamos probando es que no falla únicamente).

victor-ditay commented:
Changes done @pedrobaeza

victor-ditay commented:
@pedrobaeza

victor-ditay commented:
@pedrobaeza please review

pedrobaeza committed:
[MIG] l10n_es_aeat_mod130: Finish migration

- Remove redundant fields
- Expenses of difficult justification
- Monetary type in fields
- Adjust BOE export format
- Adjust views
- Test BOE export
- Other fine tuning

pedrobaeza reviewed (approved):
Con los últimos cambios realizados debería estar todo correcto, aunque mantengo sin fusionar hasta confirmación del cliente
